app/controllers/welcome_controller.rb

```diff
@@ -20,13 +20,16 @@ def goodbye
   end
 
   def user_home
-    bikes = current_user.bikes
-    page = params[:page] || 1
-    @locks_active_tab = params[:active_tab] == 'locks'
-    @per_page = params[:per_page] || 20
-    paginated_bikes = Kaminari.paginate_array(bikes).page(page).per(@per_page)
-    @bikes = BikeDecorator.decorate_collection(paginated_bikes)
-    @locks = LockDecorator.decorate_collection(current_user.locks)
+
+    unless return_to_if_present
```

Comment:
I've added this line so that the client is redirected to the "return_to" (doorkeeper authorization URL) after accepting the terms of service. Without it, the user will be redirected to user home  instead of continuing the oAuth 2.0 authorisation flow. That's why we are storing the "return_to" even when the client has not accept the terms of service. Let me know if it makes sense.